File: boj5766.cpp
#include <cstdio>


int main(void){
	int n, m, in, cnt[10001];
	int first=0, second=0;
	while(1){
		scanf("%d %d", &n, &m);
		if(n==0) break;
		for(int i=0; i<=10000; i++) cnt[i] = 0;
		first = second = 0;
		for(int i=0; i<n; i++){
			for(int j=0; j<m; j++){
				scanf("%d", &in);
				cnt[in]++;
				if(cnt[in] > first){
					second = first;
					first = cnt[in];	
				}
			}
		}
		for(int i=1; i<=10000; i++){
			if(cnt[i] == second)
				printf("%d ", i);
		}
		puts("");
	}
	return 0;
}

File: BOJ/problems/boj5766.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class Boj5766{
	public static BufferedReader bf = new BufferedReader( new InputStreamReader( System.in ) );
	public static StringTokenizer st = null;

	public static void main(String[] args){
		int n, m, first, second, in;
		int[] cnt;

		try {
			while(true){
				first = second = 0;
				// input n, m
				st = new StringTokenizer( bf.readLine(), " ");
				n = Integer.parseInt( st.nextToken() );
				m = Integer.parseInt( st.nextToken() );
				if(n == 0 && m == 0) break;

				cnt = new int[10001];
				for(int i=0; i<n; i++){
					st = new StringTokenizer( bf.readLine(), " ");
					for(int j=0; j<m; j++){
						in = Integer.parseInt( st.nextToken() );
						cnt[in]++;

						if(cnt[in] > first){
							second = first;
							first = cnt[in];
						}
					}
				}

				in = -1;
				for(int i=1; i<=10000; i++){
					if(cnt[i] == second){
						if(in == -1) in = i;
						else {
							System.out.print(in + " ");
							in = i;
						}
					}
				}
				if(in != -1) System.out.println(in);
			}
		} catch(IOException e){
			e.printStackTrace();
		}
		
	}

}